admin-dev/themes/new-theme/js/pages/product/edit/product-form-model.ts

```diff
@@ -88,6 +97,71 @@ export default class ProductFormModel {
     this.mapper.set(`product.${productModelKey}`, value);
   }
 
+  getTaxRatio(): BigNumber {
+    const $taxRulesGroupIdInput = this.mapper.getInputsFor('product.price.taxRulesGroupId');
+
+    if (!$taxRulesGroupIdInput) {
+      console.error('Could not find tax rules input');
+      return new BigNumber(NaN);
+    }
+
+    const $selectedTaxOption = $(':selected', $taxRulesGroupIdInput);
+    const isTaxEnabled = $taxRulesGroupIdInput.data('taxEnabled');
+
+    let taxRate = new BigNumber(0);
+
+    if (isTaxEnabled) {
+      try {
+        taxRate = new BigNumber($selectedTaxOption.data('taxRate'));
+      } catch (error) {
+        taxRate = new BigNumber(NaN);
+      }
+      if (taxRate.isNaN()) {
+        taxRate = new BigNumber(0);
+      }
+    }
+
+    return taxRate.dividedBy(100).plus(1);
```

Comment:
It feels a bit weird to divide and add something to a `BigNumber(NaN)`, BigNumber doesn't throw any error there?

Reply:
I'm not sure, it's possible that it accepts it but the value always remains NaN

But as I said in the previous comment NaN is never kept actually, it's always zero which is returned as the default fallback value so it's fine. I could perform some early return with zero though but it wouldn't reduce the code since I always need to check the value regardless because the initial input may also be invalid in the first place